client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -58,3 +58,8 @@ export function getPromotionWithLocalEdits( rootState, promotionId, siteId = get
 	return null;
 }
 
+export function getPromotionableProducts( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
```

Comment:
Does this really return promotionable products or already promoted products?

Reply:
You should be seeing your products and product categories here, for sure. Maybe re-test?

Reply:
@coderkevin It's because he picked 'product sale' which isn't implemented at this part in the stack yet.